src/Forms/Shared/Samples/Layers/ExportVectorTiles/ExportVectorTiles.xaml.cs

```diff
@@ -0,0 +1,384 @@
+        {
+            // Load the vector tile cache.
+            VectorTileCache vectorTileCache = vectorTilesResult.VectorTileCache;
+            await vectorTileCache.LoadAsync();
+
+            // Create a tile layer from the tile cache.
+            ArcGISVectorTiledLayer myLayer = new ArcGISVectorTiledLayer(vectorTileCache, vectorTilesResult.ItemResourceCache);
+
+            // Show the layer in a new map.
+            MyMapView.Map = new Map(new Basemap(myLayer));
+
+            // Re-size the mapview.
+            MyMapView.Margin = new Thickness(40);
+        }
+
+        private async Task ExportPreviewButtonTask()
+        {
+            // If preview isn't open, start an export.
+            try
+            {
+                if (!_previewOpen)
+                {
+                    // Disable the export button.
+                    MyExportPreviewButton.IsEnabled = false;
+
+                    // Save the map viewpoint.
+                    _originalView = MyMapView.GetCurrentViewpoint(ViewpointType.BoundingGeometry);
+
+                    // Start the export.
+                    await StartExport();
+                }
+                else // Otherwise, close the preview.
+                {
+                    // Change the button text.
+                    MyExportPreviewButton.Text = "Export Tiles";
+
+                    // Hide the progress bar and label.
+                    MyProgressBar.IsVisible = false;
+                    MyProgressBarLabel.IsVisible = false;
+
+                    // Clear the preview open flag.
+                    _previewOpen = false;
+
+                    // Re-size the mapview.
+                    MyMapView.Margin = new Thickness(0);
+
+                    // Re-apply the original map.
+                    MyMapView.Map = _basemap;
+
+                    // Re-apply the original viewpoint.
+                    MyMapView.SetViewpoint(_originalView);
+
+                    // Re-show the overlay.
+                    MyMapView.GraphicsOverlays.Add(_overlay);
+
+                    // Update the graphic.
+                    UpdateMapExtentGraphic();
+                }
+            }
+            catch (Exception ex)
+            {
+                await Application.Current.MainPage.DisplayAlert("Error", ex.Message, "OK");
+            }
+        }
+
+        #endregion Update Preview Map/Extent Graphic
+
+        #region EventHandlers
+
+        private void MyMapView_ViewpointChanged(object sender, EventArgs e)
+        {
+            UpdateMapExtentGraphic();
+        }
+
+        private void MyExportPreviewButton_Clicked(object sender, EventArgs e)
+        {
+            _ = ExportPreviewButtonTask();
+        }
+
+        private void MyCancelJobButton_Clicked(object sender, EventArgs e)
+        {
+            if (_job != null)
+            {
+                _ = _job.CancelAsync();
+
+                // Hide the cancel job button.
```

Comment:
I'd suggest grouping all of these lines with a single comment saying "Update the UI". I think all of the UI stuff is pretty evident from the code and doesn't need individual comments like this.